### ISSUE TITLE
Explore cautious weight decay

### ISSUE THREAD

**Issue by clessig** [ecmwf/WeatherGenerator#1650]
### Describe the task. It can be a feature, documentation, etc.

Cautious weight decay modifies standard weight decay and it reports better results:

https://arxiv.org/pdf/2510.12402

It would be worth to explore this (though no high priority). An implementation could be based on: 

https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994/3

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [x] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation
